### ISSUE TITLE
Readability Color Function

### ISSUE THREAD

**Issue by LawJolla** [styled-components/polished#185]
- `polished` version: 1.1.2
- `JSS-in_CSS` library and version: Styled-Components 2.0.0

I've gone through the docs and I don't see a function to give the most readable white or black text, e.g.

```
mostReadable('#000') // returns #fff
mostReadable('#777') // #000
```

Is this available?  If not, would you consider a PR for it?

I love polished / styled-components and thank you everyone for your hard work!

**Comment by bhough** [comment]
@LawJolla It currently is not. However, it is probably loosely related to the discussion in #148. We definitely would accept a PR for something like this as part of our V2 milestone.

**Comment by LawJolla** [comment]
@bhough sounds good.  I have a W3C readability compliant function I've been using for my projects. 

I'll get to work on it and it'll be my first PR (I read through the Contributing.MD).  Thank you for the opportunity!

**Comment by bhough** [comment]
Much appreciated @LawJolla!

A few notes since this will be going into V2 and we are just ramping that up.

1. Create your branch/pr from https://github.com/styled-components/polished/tree/version-2 as that has the latest build setup.
2. If any of the CSS props are hyphenated normally, be sure to camelCase them instead, as that will be required once #170 is merged into v2 (hopefully very soon)
3. Before we merge it, we will need to integrate the new messageHandler we are implementing in V2. I'll hit you up with more details once that is ready to go.

Thanks again for taking the time to contribute.

**Comment by LawJolla** [comment]
My first crack at it is ready to go when you are.  Just let me know.  Thanks!

**Comment by bhough** [comment]
Thanks LawJolla! #170 is in review, so that should be any day now. 3 on the list above is coming along nicely, hopefully sometime this week, then I can walk you through the new api for module validation we adding to ensure pretty warnings, error isolation, and reduce the amount of invalid generated CSS.

**Comment by bhough** [comment]
@LawJolla I've managed to get the new validation API done in a way that shouldn't directly affect your code. If you want to submit a WIP PR we can start to take a look at things.

**Comment by LawJolla** [comment]
Thanks @bhough !  #189 is submitted!  Thanks for your time and review.  I think it's a good contribution.

As an aside, I just saw you're with BR.  My backend stack is Elixir/Absinthe.  Good stuff!

**Comment by bhough** [comment]
Thanks @LawJolla I will give it a look this weekend. 

Nice, we're obviously huge into Elixir at BR, so I approve 😬

**Comment by bhough** [comment]
Addressed by #192 woot! 🎉
